src/python/python/ad/reparam.py

```diff
@@ -96,7 +96,7 @@ def _sample_warp_field(scene: mi.Scene,
 
     # Convert into a direction at 'ray.o'. When no surface was intersected,
     # copy the original direction
-    hit = si.is_valid()
+    hit = si.is_valid() & dr.any(dr.neq(si.p, ray.o))
```

Comment:
I am curious, why is this necessary? Is it specifically handling self-intersection with curves?

Btw, I wonder whether it wouldn't be necessary to use some epsilon.